godal.go

```diff
@@ -3814,6 +3814,51 @@ func GridCreate(pszAlgorithm string,
 	return nil
 }
 
+// Grid runs the library version of gdal_grid.
+// See the gdal_grid doc page to determine the valid flags/opts that can be set in switches.
+//
+// Example switches :
+//
+//	[]string{"-a", "maximum", "-txe", "0", "1"}
+//
+// Creation options and driver may be set in the switches slice with
+//
+//	switches:=[]string{"-co","TILED=YES","-of","GTiff"}
+//
+// NOTE: Some switches are NOT compatible with this binding, as a `nullptr` is passed to a later call to
+// `GDALGridOptionsNew()` (as the 2nd argument). Those switches are: "-oo", "-q", "-quiet"
+func (ds *Dataset) Grid(destPath string, switches []string, opts ...GridOption) (*Dataset, error) {
+	gridOpts := gridOpts{}
+	for _, opt := range opts {
+		opt.setGridOpt(&gridOpts)
+	}
+
+	// Throw error for any invalid flags
+	var badSwitches []string
+	for _, s := range switches {
+		if s == "-oo" || s == "-q" || s == "-quiet" {
+			badSwitches = append(badSwitches, s)
+		}
+	}
+	if len(badSwitches) > 0 {
+		return nil, fmt.Errorf("one or more invalid switch values provided: %s", strings.Join(badSwitches, ","))
+	}
```

Comment:
```suggestion
```
we don't do any such validation for the other utilities we wrap, let's not start doing so here.